### PR TITLE
:has() selector (Positive)

### LINKED ISSUE #528: :has() pseudo class
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: The Relational Pseudo-class: ':has()'
* Specification or proposal URL: https://www.w3.org/TR/selectors-4/#relational
* Caniuse.com URL (optional): https://caniuse.com/css-has
* Bugzilla URL (optional): https://bugzilla.mozilla.org/show_bug.cgi?id=418039
* Mozillians who can provide input (optional): 

### Other information

The :has() pseudo class is a selector that specifies elements which has at least one element that matches the relative selector passed as an argument.

Unlike other selectors, it gives a way to apply a style rule to preceding elements (preceding siblings / ancestors / preceding siblings of ancestors) of a certain elements.

This difference is attractive to web developers, but also it generates lots of concerns mainly about performance and complexity.

We thought that, clarifying those concerns would be helpful for the discussion. So we started to check feasibility on blink, and were able to get some meaningful and reasonable results from this step. Based on the results, we are going to move forward by prototyping.

You can get the details about the current status in here.
* Explainer: https://github.com/Igalia/explainers/tree/main/css/has
* Test summary: https://css-has.glitch.me/

### PR DESCRIPTION
**Pull request by dshin-moz**
Closes #528

### CODE REVIEW

File: activities.json [hunk 1]
```diff
@@ -306,6 +306,19 @@
...
     "url": "https://drafts.css-houdini.org/css-properties-values-api-1#at-property-rule"
   },
+  {
+    "ciuName": "css-has-selector",
+    "description": ":has selector lets authors select elements that anchor at least one element that matches its inner relative selector.",
+    "id": "css-has-selector",
+    "mdnUrl": "https://developer.mozilla.org/en-US/docs/Web/CSS/:has",
+    "mozBugUrl": "https://bugzilla.mozilla.org/show_bug.cgi?id=has-pseudo",
+    "mozPosition": "positive",
+    "mozPositionDetail": "We acknowledge the developer demand for this selector, and the implementation is shipping as of 121. However, we are still concerned about the worst-case performance of DOM tree mutation when using the selector.",
```
Comment (zcorpan) [inline activities.json:316]: I think we generally try to not reference Firefox shipping status in standards-positions rationale. Try to answer if the feature is good for the web.

Also, I think this comes across as basically negative. "We had to implement it, but we don't like it."

**Review by zcorpan — CHANGES_REQUESTED** [review]

**Review by tantek — APPROVED** [review]
LGTM, verified this works

**Review by tantek — APPROVED** [review]
LGTM with the patch updates here.

File: activities.json (hunk 1 above)
Comment (martinthomson) [reply activities.json:316]: How about:

```suggestion
    "mozPositionDetail": "We recognize that :has is something that a lot of people want. That power does come with performance costs, like the potential the worst-case performance resulting from DOM tree mutation to be severe. Overall, the utility of the selector justifies this risk, but we might need to do more to help developers avoid the worst problems.",
```

**Review by tantek — APPROVED** [review]
LGTM with @martinthomson's suggested change.

**Review by zcorpan — APPROVED** [review]

**Review by czarnajusta23-boop — COMMENTED** [review]
activities.json6

**Pull request merged**